tests/integration/test_vulnerability_detector/test_feeds/test_invalid_values_redhat_feeds.py

```diff
@@ -0,0 +1,151 @@
+]
+
+# Configuration data
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+
+
+# Fixtures
+@pytest.fixture(scope='module', params=configurations, ids=ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture()
+def modify_feed(field_info, custom_input, request):
+    """
+    Modify the redhat OVAL feeds, setting a test field value
+    """
+    backup_data = json.loads(read_file(custom_redhat_oval_feed_path))[0]
+
+    modified_data = dict(backup_data)
+
+    modified_data[field_info['field']] = custom_input
+
+    write_file(custom_redhat_oval_feed_path, json.dumps([modified_data], indent=4, ensure_ascii=False))
+
+    clean_vuln_and_sys_programs_tables()
+
+    control_service('restart', daemon='wazuh-modulesd')
+
+    set_custom_system(os_name=system_data['os_name'], os_major=system_data['os_major'],
+                      os_minor=system_data['os_minor'], name=system_data['name'])
+
+    yield
+
+    write_file(custom_redhat_oval_feed_path, json.dumps([backup_data], indent=4))
+
+    clean_vuln_and_sys_programs_tables()
+
+    truncate_file(LOG_FILE_PATH)
+
+
+def check_feed_imported_successfully():
+    """
+    Check that redhat OVAL feeds have been imported successfully
+    """
+    check_vulnerability_event(
+        wazuh_log_monitor=wazuh_log_monitor, update_position=False, timeout=10,
+        callback=expected_success_message,
+        error_message=f"Could not find the message: 'Red Hat Enterprise Linux' feed finished successfully"
+    )
+
+    check_vulnerabilities_number(expected_number=5)
+
+
+def check_failure_when_importing_feed():
+    """
+    Check an error message when importing redhat OVAL feeds and checks that the vulnerabilities table is empty
+    """
+    check_vulnerability_event(
+        wazuh_log_monitor=wazuh_log_monitor, update_position=False, timeout=10,
+        callback=expected_error_message,
+        error_message=f"Could not find the message: CVE database could not be updated"
+    )
+
+    check_vulnerabilities_number(expected_number=0)
+
+
+@pytest.mark.parametrize('field_info, custom_input', itertools.product(field_data, inputs))
+def test_invalid_redhat_feed(field_info, custom_input, get_configuration, configure_environment, restart_modulesd,
+                             modify_feed):
+    """
+    Check if vulnerability detector behaves as expected when importing redhat OVAL feeds with wrong field values
+    """
+    if field_info['field'] == 'affected_packages':
+        pytest.xfail("Add error messages to this case use. Issue: https://github.com/wazuh/wazuh/issues/5151")
+
+    if field_info['field'] in REDHAT_KEY_FIELDS_FEEDS:
+        if str(type(custom_input)) == field_info['type']:
+            check_feed_imported_successfully()
+        else:
+            check_failure_when_importing_feed()
+    else:
+        check_feed_imported_successfully()
```

Comment:
```suggestion
    if field_info['field'] in REDHAT_KEY_FIELDS_FEEDS and not type(custom_input) is field_info['type']::
            check_failure_when_importing_feed()
    else:
        check_feed_imported_successfully()
```

Reply:
Done in ab1c229f9625689d173018ae41d5437228ded6be